### DISCUSSION TITLE
Can you specify multiple ports to open and listen to rtsp stream?

### DISCUSSION THREAD

**Discussion by gj-raza** [bluenviron/mediamtx#2932] [Questions]
### Question

https://github.com/bluenviron/mediamtx/blob/fd2466a8e5bf4e1e23857202967d1dd1416ce841/mediamtx.yml#L88

i am looking to create a test setup in which i want to stream multiple video files via ffmpeg to the server each at different port. Both the server and ffmpeg stream commands will be running on localhost. can i specify multiple ports to listen to rtsp streams in the config file? if not, how can i achieve this?

I am using this ffmpeg command to publish streams:
ffmpeg -re -stream_loop -1 -i file.ts -c copy -f rtsp rtsp://localhost:8554/mystream

**Comment by cubisttriangle** [comment]
Port 8554 is the default RTSP listener port. The server can handle multiple clients publishing streams and multiple clients requesting streams. You don't need to specify multiple ports, you need to specify different stream paths.

For example:
  ... rtsp://localhost:8554/mystream_0
  ... rtsp://localhost:8554/mystream_1

**Reply by cloudchef-wyatt** [reply]
Following up, while it may support multiple streams on the same port, is there a way to stream RTSP to different ports?
There are two pieces to this question
1. Explicitly, is it possible.
2. Implicitly, is there a performance difference between streaming to one port vs multiple ports, and if there isn't, is it due to the OS or is it abstracted away by mediamtx?

**Reply by cubisttriangle** [reply]
I think you might be confused about how RTSP works. The RTSP port is just a port where commands are sent. These commands do things like negotiate ports for video streams (which will be different than the RTSP port).

**Reply by cloudchef-wyatt** [reply]
Perfect, thank you for the explaination @cubisttriangle.
